### PR TITLE
Build mimic with cores determined by memory calc

### PR DESCRIPTION
**Pull request by el-tocino**
Use system memory to calculate number of cores to build with. Pass as a parameter to the install scripts. 

zeehio indicated that each core takes ~460mb of RAM to build. For padding, used 500mb to calculate number of cores. If sysmem divided by 500mb is less than nproc, then set CORES less than nproc. On a 12gb machine, for instance, it indicates max 24 cores.  System has 8 (4+hyperthreading), so it set CORES to 8.  On a pi3, with 128mb set for gpu mem, it calculates 1 (917504/500000 = 1), as we would prefer.  

The hash command can break the -Ee settings when it doesn't find a copy of mimic, so bypass the exit on error when checking for that and re-enable error bombing after.

### CODE REVIEW

File: dev_setup.sh [hunk 1]
```diff
@@ -61,12 +61,12 @@ if [ ! -d "${VIRTUALENV_ROOT}" ]; then
 fi
 source "${VIRTUALENV_ROOT}/bin/activate"
 cd "${TOP}"
-easy_install pip==7.1.2 # force version of pip
-pip install --upgrade virtualenv
+#easy_install pip==7.1.2 # force version of pip
+#pip install --upgrade virtualenv
```
Comment (forslund) [inline dev_setup.sh]: Did you mean to have these commented out?

File: dev_setup.sh [hunk 2]
```diff
@@ -61,12 +61,12 @@ if [ ! -d "${VIRTUALENV_ROOT}" ]; then
...
 
 # install requirements (except pocketsphinx)
 # removing the pip2 explicit usage here for consistency with the above use.
-pip install -r requirements.txt 
+#pip install -r requirements.txt
```
Comment (forslund) [inline dev_setup.sh]: and this?

**Comment by el-tocino** [comment]
@forslund no, had them commented to test the math and script paramter bits and have now reverted that.

**Comment by forslund** [comment]
Hi @el-tocino 
The calculation looks okey but the removed error check seems a bit weird. Will we notice if mimic is broken in this setup?

I started a rebase of this (excluding the msm fix) in https://github.com/MycroftAI/mycroft-core/tree/eltocino-mimic-cpu-calc-rebase where I removed the `set +eE` and Travis doesn't seem to have a problem with this: https://travis-ci.org/MycroftAI/mycroft-core/builds/262208211

Is the rebased branch OK for you?

**Comment by el-tocino** [comment]
I had it turned off there for that since it was exiting when mimic wasn't found.  If it doesn't fail when running on a host that does not have mimic installed it should be ok.  Will verify a bit later.

**Comment by el-tocino** [comment]
So that rebase worked.  I think that'll be fine then.  I also think we can skip the output redirect (2&>1) on the hash, it shows the error but doesn't fail, which seems ok.  I can move the msm changes to a new pr if you'd prefer, will close this one.

**Pull request closed without merging**